Fixed incompatible plugin text typo

paulopmt1 opened the pull request:
#### Changes proposed in this Pull Request

* Fixed incompatible plugin text typo
* Added a new rule for showing incompatible message to self hosted sites

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open the /plugin page and search for `wordfence`, you should see the correct text `Why is this plugin not compatible with WordPress.com?` instead of `Why this plugin is not compatible with WordPress.com?`
* Make sure the message only appears for Atomic sites with Jetpack, and Simple sites
* Create a new jurassic.ninja site, enable Jetpack connection and make sure you don't see the message

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Thanks for finding it @allilevine!

Related to https://github.com/Automattic/wp-calypso/issues/62644

paulopmt1 committed:
Fixed incompatible plugin text typo

allilevine reviewed (dismissed):
The typo is fixed with this change ✅

paulopmt1 committed:
Updated rules to when show incompatible plugins message:
- Will only show for Jetpack sites AND
- Will only show for Atomic sites hosted by wordpress.com

allilevine committed:
Check for nonAtomic Jetpack sites

allilevine reviewed (approved):
I'm seeing the fixed typo and the graying out only happens for Atomic and Simple sites, not Jetpack self-hosted.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7231940

Hi @paulopmt1, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Why is this plugin not compatible with WordPress.com?`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7231940).
